ports/servo/glutin_app/window.rs

```diff
@@ -1448,3 +1466,23 @@ fn filter_nonprintable(ch: char, key_code: VirtualKeyCode) -> Option<char> {
         None
     }
 }
+
+fn sanitize_url(request: &str) -> Option<ServoUrl> {
+    let request = request.trim();
+    ServoUrl::parse(&request).ok()
+        .or_else(|| {
+            // FIXME(paul): this is way too naive.
+            // We want to know if only the scheme is missing.
+            // See: https://github.com/paulrouget/servoshell/issues/59
+            if request.ends_with(".com") || request.ends_with(".org") || request.ends_with(".net") {
```

Comment:
I think `request.contains('/') || is_reg_domain(request)` from net_traits/pub_domains.rs will probably do what you want?